Merged and reworked episode 03

anenadic opened the pull request:
Merged Anne's and mine changes in episode 03, but then pulled out my bits on website configuration and simply pasted in episode 05 on a [separate branch](https://github.com/carpentries-incubator/building-websites-with-jekyll-and-github-or-gitlab/tree/episode-05) (which will now need reworking too). I have tested the code yet - this will need reviewing once again, when we have most of the content and the whole narrative ready.

anenadic committed:
Merged and reworked episode 03

annefou commented on _episodes/03-github-pages.md:
```diff
@@ -3,275 +3,126 @@ title: "Hosting Pages and Websites on GitHub"
+There are two approaches to tell GitHub your repository is hosting a website and configure GitHub pages to render it. 
 
-It is a convention used by Github and we need to change this default behavior because the content of our repository is under a branch called *main*.
+The first one is to create a new branch called `gh-pages`. By convention, GitHub Pages uses this branch to look for the website content and by creating it you implicitly tell GitHub that you want your content published. Once you can create `gh-pages` from your current branch (`main`, created by default when you created the repository), you can then choose to delete the branch `main` to avoid any  ...
```
I wonder if it makes sense to talk about gh-pages at this stage. 

Wouldn't be simpler to show how to select `main` branch to "create" a website and than have a called out about gh-pages.

anenadic replied:
Yes, might have got things backward - I tell them about gh-pages and then go and explain the approach which is actually less favourable. I'll fix this tomorrow as per your suggestion @annefou.

anenadic replied:
OK, I fixed the text to address this comment @annefou via ec981b4891081876fffcc766ddc2ba1439c0226b.

annefou reviewed (commented):
All other changes are great. So I don't know what you feel about my comment about `gh-pgaes` but maybe we can merge fierst and re-discuss it later.

anenadic committed:
Better explanation on configuring GH Pages as per https://github.com/carpentries-incubator/building-websites-with-jekyll-and-github-or-gitlab/pull/52#pullrequestreview-503311241

anenadic committed:
Fixed wording

anenadic committed:
Fixed wording - part 2

anenadic committed:
Fixed wording - part 3

tobyhodges commented on _episodes/03-github-pages.md:
```diff
@@ -3,275 +3,126 @@ title: "Hosting Pages and Websites on GitHub"
+There are other services available to create and publish website but one of the main advantages of GitHub Pages is that you can version control your website and therefore keep track of all your changes. This is particularly helpful for collaborating on a project website. [GitLab](https://about.gitlab.com/) offers very similar services but GitHub pages is the simplest approach.
 
-## Turn on GitHub Pages for your main branch
+Let's continue from the GitHub repository we have created in the previous episode. One important file you should have already is `index.md`: it will become the homepage of your project website.
```
I think the learners will only have the `README.md` file at the starting point of this lesson. In the absence of a file called `index.md`, the README is used as the landing page of the GitHub Pages site, so it shouldn't change anything for this section, except adding a part to create a separate `index.md` later...

anenadic replied:
Oh, I've missed that bit of discussion and have just added exercises that work with `index.md`. It's not a biggie, I will  add a bit where we create it.

anenadic replied:
Fixed via 596a28b.

tobyhodges commented on _episodes/03-github-pages.md:
```diff
@@ -3,275 +3,126 @@ title: "Hosting Pages and Websites on GitHub"
+    ![Project website URL - successfull build](../fig/website_url_green_tick.png)
 
-![Default project website](../fig/default_website.png)
+5. Once ready, you should see the contents of the `index.md` file that we created earlier, rendered as a website. 
```
```suggestion
5. Once ready, you should see the contents of the `README.md` file that we created earlier, rendered as a website. 
```

anenadic replied:
Fixed via 596a28b.

tobyhodges commented on _episodes/03-github-pages.md:
```diff
@@ -3,275 +3,126 @@ title: "Hosting Pages and Websites on GitHub"
-> Add a new section **About** in `index.md`. You can edit `index.md` from the GitHub interface and directly commit your changes.
-> Then view your changes on the website.
+By doing this, you add a link to the repository's website on your repository's landing page and anyone (including yourself) can access it quickly when visiting your GitHub repository.
+
```
```suggestion

Up to now, the content of your webpage is identical to what visitors to your repository on GitHub will see in the project's `README.md` file. It is often better to have different content in the homepage of your site - aimed at visitors to the website - than in the README, which is supposed to provide information about the GitHub repository e.g. license information, local installation instructions, the structure and configuration of the repository, etc. By default, the homepage for a GitHub Pages website is built from a file called `index.md`: in the absence of a file with that name the "fallback" option is to use `README.md`, which is why your homepage is currently being built from that file.

To separate the contents of the repository's README from the website's homepage, create a new file called `index.md` and copy the contents of the README into that file. We'll add more content to that file in the next exercise.
```

Proposing this to bridge the gap between README and `index.md` for the homepage. Feel free to overwrite/ignore any of it!

tobyhodges reviewed (changes requested):
At the sprint yesterday, I think we concluded that all exercises in the Markdown episode would be done in `README.md`, meaning there won't yet be an `index.md` file by the beginning of this episode. (@JulianKarlBauer can confirm...) I've suggested an additional paragraph to bridge this gap, but you might want to do it differently...

anenadic committed:
Added text and screenshots for exercises

anenadic committed:
Update _episodes/03-github-pages.md

Added Toby's suggestion to bridge from README.md being the default landing page to having an index.md.

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic committed:
Addressed the non existence of index.md at the start of the episode - see discussion https://github.com/carpentries-incubator/building-websites-with-jekyll-and-github-or-gitlab/pull/52#pullrequestreview-503644312

anenadic commented:
The episode is ready for review once again - I have added solutions to exercises in the meantime and addressed the fact that we only have README.md at the start of the episode @tobyhodges @annefou.

tobyhodges commented:
Looks great, thank you @anenadic

tobyhodges commented on _episodes/03-github-pages.md:
```diff
@@ -3,275 +3,182 @@ title: "Hosting Pages and Websites on GitHub"
+Now that you know how to create Markdown files, let's see how to turn them into web pages. GitHub has a 
+service just for that called GitHub Pages.  
+
+# Publishing a Repository as a Website With GitHub Pages
```
```suggestion
# Publishing a Website with GitHub Pages
```

tobyhodges commented on _episodes/03-github-pages.md:
```diff
@@ -3,275 +3,182 @@ title: "Hosting Pages and Websites on GitHub"
-> ## Why Github?
->
-> There are other services available to create and publish website but one of the main advantage of GitHub Pages is that you can version control your website and therefore keep track of all your changes. This is particularly helpful for collaborating on a project website. [GitLab](https://about.gitlab.com/) offers very similar services but GitHub pages is the simplest approach.
+## Turning on GitHub Pages for Your Repository
```
```suggestion
## Enabling GitHub Pages
```

tobyhodges commented on _episodes/03-github-pages.md:
```diff
@@ -3,275 +3,182 @@ title: "Hosting Pages and Websites on GitHub"
+{: .callout}
 
-![Default project website](../fig/default_website.png)
+## Updating Repository Details With Website URL
```
```suggestion
## Making Your Pages More Findable
```

tobyhodges reviewed (commented):
Putting some suggestions here for the headings, to address #57. Happy to prepare these in a separate PR if you prefer?

anenadic committed:
Update _episodes/03-github-pages.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic committed:
Update _episodes/03-github-pages.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic committed:
Update _episodes/03-github-pages.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic commented:
> Putting some suggestions here for the headings, to address #57. Happy to prepare these in a separate PR if you prefer?

All fab, thank you! Accepted them and will close the issue #57. Ready for merge I'd say @tobyhodges ?

tobyhodges commented:
Thanks @anenadic 😀
